packages/zoe/src/contractFacet/types.js

```diff
@@ -98,6 +99,35 @@
  * @returns {Promise<Invitation>}
  */
 
+/**
+ * @typedef {Object} InvitationWithSeat
+ * @property {Promise<Invitation>} invitation
+ * @property {ZCFSeat} zcfSeat
+ * @property  {ERef<UserSeat>} userSeat
```

Comment:
Actually this is a terrible security flaw, as this would allow the contract to get a payout before the user. We would break offer safety and payout liveness!